vnext/Microsoft.ReactNative/Fabric/Composition/Modal/WindowsModalHostViewComponentView.cpp

```diff
@@ -11,23 +10,158 @@
+// SendMessage)
+LRESULT CALLBACK LogBoxWndProc(HWND hwnd, UINT message, WPARAM wparam, LPARAM lparam) noexcept {
+  auto data = reinterpret_cast<::IUnknown *>(GetProp(
+      hwnd,
+      CompHostProperty)); // gets data handle from the property list of specified window (ie the window we want to make)
+  winrt::com_ptr<winrt::IUnknown> spunk;
+  winrt::Microsoft::ReactNative::CompositionHwndHost host{nullptr};
+
+  if (data) {
+    winrt::check_hresult(data->QueryInterface(
+        winrt::guid_of<winrt::Microsoft::ReactNative::CompositionHwndHost>(),
+        winrt::put_abi(host))); // look into the data for a CompositionHwndHost and store it in host
+    auto result = static_cast<LRESULT>(host.TranslateMessage(message, wparam, lparam));
+    if (result)
+      return result;
+  }
+
+  switch (message) {
+    case WM_NCCREATE: { // sent before WM_CREATE, lparam should be identical to members of CreateWindowEx
+      auto createStruct = reinterpret_cast<CREATESTRUCT *>(lparam); // CreateStruct
+      auto windowData = static_cast<::IUnknown *>(createStruct->lpCreateParams);
+      SetProp(hwnd, CompHostProperty, reinterpret_cast<::IUnknown *>(windowData)); // adds new properties to window
+      break;
+    }
+    case WM_CREATE: { // recieves after window is created but before visible
+      // host.Initialize((uint64_t)hwnd); // this line currently breaks Modal and LogBox
+      break;
+    }
+    case WM_CLOSE: {
+      // Just hide the window instead of destroying it
+      ::ShowWindow(hwnd, SW_HIDE);
+      return 0;
+    }
+    case WM_DESTROY: { // called when we want to destroy the window
+      data->Release();
+      SetProp(hwnd, CompHostProperty, nullptr);
+    }
+  }
+
+  return DefWindowProc(hwnd, message, wparam, lparam);
+}
+
+// Creates and Register a new window class
+void WindowsModalHostComponentView::RegisterWndClass() noexcept {
+  static bool registered = false;
+  if (registered)
+    return;
+
+  HINSTANCE hInstance =
+      GetModuleHandle(NULL); // returns a handle to the file used to create the calling process (.exe file)
+
+  WNDCLASSEXW wcex = {}; // contains window class information
+  wcex.cbSize = sizeof(WNDCLASSEX); // size of windows class (bytes)
+  wcex.style = CS_HREDRAW | CS_VREDRAW; // class style (redraw window on size adjustment)
+  wcex.lpfnWndProc = &LogBoxWndProc; // pointer to windows procedure
+  wcex.cbClsExtra = DLGWINDOWEXTRA; // extra bytes to allocate
+  wcex.cbWndExtra =
+      sizeof(winrt::impl::abi<winrt::Microsoft::ReactNative::ICompositionHwndHost>::type *); // extra bytes to allocate
+  wcex.hInstance = hInstance;
+  wcex.hCursor = LoadCursor(nullptr, IDC_ARROW); // handle to class cursor
+  wcex.hbrBackground = (HBRUSH)(COLOR_WINDOW + 1); // background color
+  wcex.lpszClassName = c_modalWindowClassName; // specify resource name
+  ATOM classId = RegisterClassEx(&wcex); // register new windows class
+  WINRT_VERIFY(classId); // 0 = fail
+  winrt::check_win32(!classId);
```

Comment:
Is `WINRT_VERIFY` a debug-only assert? [looks like no](https://learn.microsoft.com/en-us/windows/uwp/cpp-and-winrt-apis/macros#winrt_assert-winrt_verify)
Why do we have both checks here?

`check_win32` takes in a... success code? [reference](https://learn.microsoft.com/en-us/uwp/cpp-ref-for-winrt/error-handling/check-win32). But this an ATOM. Doesn't look like it does the work of calling GetLastError? Basically... I'm not sure this is the right error checking.

Reply:
huh interesting! I'm not sure on the right error checking for an ATOM but I can do some research! (unless someone else knows)

Doing a quick search of our repo, https://sourcegraph.com/search?q=context:global+repo:%5Egithub%5C.com/microsoft/react-native-windows%24+RegisterClassEx&patternType=keyword&sm=0, it looks like this is how we check for an error when calling RegisterClassEX in multiple place, wondering if we will also want to change these?